packages/loader/container-loader/src/loader.ts

```diff
@@ -202,10 +203,60 @@ export interface ILoaderProps {
 	readonly protocolHandlerBuilder?: ProtocolHandlerBuilder;
 }
 
+/**
+ * Other props which are used to load a container on top of the basic loader props.
+ * @alpha
+ * @legacy
+ */
+export interface IResolveContainerProps extends ILoaderProps {
+	/**
+	 * The request to resolve the container.
+	 */
+	request: IRequest;
+
+	/**
+	 * Pending local state to be applied to the container.
+	 */
+	pendingLocalState?: string;
+}
+
+/**
+ * Other props which are used to create a detached container on top of the basic loader props.
+ * @alpha
+ * @legacy
+ */
+export interface ICreateDetachedContainerProps extends ILoaderProps {
+	/**
+	 * The code details for the container to be created.
+	 */
+	codeDetails: IFluidCodeDetails;
+
+	canReconnect?: boolean;
```

Comment:
Here and elsewhere add `| undefined` to optional types to align with [exactOptionalPropertyTypes](https://www.typescriptlang.org/tsconfig/#exactOptionalPropertyTypes)